### DISCUSSION TITLE
Directory for importing arbitrary code?

### DISCUSSION THREAD

**Discussion by vzakharov** [nuxt/nuxt#18573] [Questions]
I’m building some core functionality that I will later migrate to a separate package, but at the moment I want the simplicity of being able to import it from a Nuxt project that has the necessary UI for testing, admin, etc.

I tried putting the files into an arbitrary directory (`~/core` to be precise), but Nuxt doesn’t seem to find them there, because it gives a `Cannot find module` error when I try to `import Something from '~/core/something'`.

Is there some special place I can put such modules to? I see something about `utils` in the docs, but the documentation is very scarce. One thing it does mention is “auto-imports,” which I definitely do NOT want for these modules.

Any help appreciated.

P.S. Don’t know if it’s of any use, but the imports work fine from within, say, `/server/api/[...].ts`, but not from `pages`.

**Comment by TheAlexLichter** [comment]
"Classic" imports should work fine from whatever folder you like as you'd do it in other JS/TS projects. See [this example](https://stackblitz.com/edit/nuxtdiscussions18573?file=core%2Fadd.ts,app.vue).
"Auto-imports" are only for a subset of folders (but more can be added manually too) - but as you mentioned this is not wanted in your case.

**Reply by vzakharov** [reply]
> "Classic" imports should work fine from whatever folder you like as you'd do it in other JS/TS projects. See [this example](https://stackblitz.com/edit/nuxtdiscussions18573?file=core%2Fadd.ts,app.vue).

Yeah, that’s what I thought, except for me, it doesn’t.

Not working from a page (see directory structure on the left, selected import on the right, error message in tooltip):

![image](https://user-images.githubusercontent.com/188197/215083906-e920e6bb-a64b-4149-bd1b-3631dd83afeb.png)

Working fine from within /server:

![image](https://user-images.githubusercontent.com/188197/215083866-257a204b-9690-4e6d-afda-1541b2973fd6.png)

**Reply by vzakharov** [reply]
As a workaround, I changed `/core` to `/utils/core`, and it does seem to work, but this somehow doesn’t seem an optimal solution to me (auto imports but not only).

**Reply by TheAlexLichter** [reply]
Strange! Can you check if you can [reproduce](https://nuxt.com/docs/community/reporting-bugs/#create-a-minimal-reproduction) the error somehow? 
Also, you can try `@` or `@@`/`~~` which should work too (the later rare for the "project root" which is the same as ~/@ by default).

Did you alter the tsconfig?
